include/ginkgo/core/solver/ir.hpp

```diff
@@ -212,6 +188,9 @@ class Ir : public EnableLinOp<Ir<ValueType>>, public Transposable {
     void apply_impl(const LinOp* alpha, const LinOp* b, const LinOp* beta,
                     LinOp* x) const override;
 
+    void set_relaxation_factor(
+        std::shared_ptr<const matrix::Dense<ValueType>> new_factor);
```

Comment:
Do we need this to take a `Dense` ? It might be better to take a scalar value ?

Reply:
Let's address this together with #820